### ISSUE TITLE
After a couple of months I stop downloading all the tweets...

### ISSUE THREAD

**Issue by jorge-datanalyst** [twintproject/twint#657]
# Issue Template
Please use this template!

### Initial Check
> If the issue is a request please specify that it is a request in the title (Example: [REQUEST] more features). If this is a question regarding 'twint' please specify that it's a question in the title (Example: [QUESTION] What is x?). Please **only** submit issues related to 'twint'. Thanks.

>Make sure you've checked the following:

- [] Python version is 3.6.2;
- [] Updated Twint with `pip3 install --user --upgrade -e git+https://github.com/twintproject/twint.git@origin/master#egg=twint`;
- [] I searched for the problem but I can't find an answer.

### Command Ran
>
d = datetime.date.today()
dt = d - timedelta(days=1)
d=str(d)
dt=str(dt)

c = twint.Config()
c.Since = dt
c.Search = "GDP"
c.Store_csv = True
c.Count = True
c.Limit = 10000
c.Hide_output = True
twint.run.Search(c)

>some examples of queries

"from:@bethanymac12 OR rt:@bethanymac12 OR from:@grlemkau OR rt:@grlemkau OR from:@HayekAndKeynes OR rt:@HayekAndKeynes OR from:@LizAnnSonders OR rt:@LizAnnSonders OR from:@russian_market OR rt:@russian_market OR from:@BullandBaird OR rt:@BullandBaird OR from:@JustinWolfers OR rt:@JustinWolfers OR from:@PaulKrugman"

"FOMC OR BCE OR BOJ OR BOE OR FED OR (Federal Reserve) OR Yellen OR Draghi OR Carney OR (Repo rate) OR (quantitative easing) OR tapering OR (fed funds rate)"

"(Volcker rule) OR FATCA OR (Sarbanes oxley)"

"EUR OR JPY OR GBP OR (currency wars) OR BRL OR (TRY turkey) OR CLP OR MXN OR COP"

### Description of Issue
>

My code was working without any problem from November 25, 2019 until February 1, 2020, from February 2 I stop downloading all the tweets, the only thing I detect with the naked eye is that the script works well when it should Download few tweets and it starts to fail when you should download larger amounts and just after throwing the following error:

query: GDP
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: Tweets_known_error: Expecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
Expecting value: line 1 column 1 (char 0) [x] run.Feed
[!] if you get this error but you know for sure that more tweets exist, please open an issue and we will investigate it!
[+] Finished: Successfully collected 8699 Tweets.

Another strange thing is that before when it worked without problems it also threw the same error in many occasions but without causing problems in the download. Now after throwing this error the script continues to run but with zero downloads:

Query: (Industrial Production) OR PMI OR (Manufacture data)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: Tweets_known_error: Expecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
Expecting value: line 1 column 1 (char 0) [x] run.Feed
[!] if you get this error but you know for sure that more tweets exist, please open an issue and we will investigate it!
[+] Finished: Successfully collected 0 Tweets.

Query: FOMC OR BCE OR BOJ OR BOE OR FED OR (Federal Reserve) OR Yellen OR Draghi OR Carney OR (Repo rate) OR (quantitative easing) OR tapering OR (fed funds rate)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: Tweets_known_error: Expecting value: line 1 column 1 (char 0)
CRITICAL: root: twint.run: Twint: Feed: noDataExpecting value: line 1 column 1 (char 0)
Expecting value: line 1 column 1 (char 0) [x] run.Feed
[!] if you get this error but you know for sure that more tweets exist, please open an issue and we will investigate it!
[+] Finished: Successfully collected 0 Tweets.

### Environment Details
>Using Linux, Ubuntu 18.04.3 LTS, Running this in Anaconda, with Jupyter Notebook and also in 
 script.py?

Transient hostname: li167-249
Icon name: computer-vm
Chassis: vm
Virtualization: kvm
Operating System: Ubuntu 18.04.3 LTS
Kernel: Linux 4.15.0-50-generic
Architecture: x86-64

Sorry for my bad English, my native language is Spanish and I don't speak English well.

**Comment by pielco11** [comment]
Twint version?

**Comment by jorge-datanalyst** [comment]
tested with 2.1.13 and 2.1.6 and has the same result

**Comment by pielco11** [comment]
If you search with the browser, do you get results?

**Comment by jorge-datanalyst** [comment]
Hello, if I have searched every query I entered in the twitter browser and if it shows results.

My persecution is that the TWINT module after making a download more or less large as about 10,000 tweets approximately is unconfigured or something happens because after this all the queries that I enter give zero.

after a while if I restart the search and restart the kernel of the juoyter which is where I do the tests and start in one of the queries that had thrown me zero then it starts well, throws the downloads with results and after this it continues throwing zero with the other queries that you indicate.

These tests I have also done with a script.py to see if the problem is with the jupyter but exactly the same happens

**Comment by pielco11** [comment]
Twitter is applying workarounds to limit Twint's functionality

**Comment by jorge-datanalyst** [comment]
Hello @pielco11, first of all thank you very much for your answers, on the other hand I am very sorry to know that, I have my project supported in TWINT. Do you know of any alternative or way to avoid these limitations, would any suggestions be helpful?

**Comment by pielco11** [comment]
I don't know, I tried a couple of your queries and those seem to be running, I get the tweets

For example

![image](https://user-images.githubusercontent.com/17363750/74148235-daa04600-4bfc-11ea-93dd-d3300e90e196.png)

**Comment by jorge-datanalyst** [comment]
Hello, if it works many times but for example my script has a loop with approximately 70 queries when I start to pass the queries for TWINT to do the downloads it does well until it reaches a query that has many tweets like about 8,000 or something like that, at this point it throws the errors that you send me before and makes the download but after this all the downloads go to zero. As I mentioned before this did not happen but from the 2 d February yes.

**Comment by nonameable** [comment]
@pielco11 Why do you think Twitter is applying workarounds to limit Twint's functionality? Maybe other similar issues lead to the same conclusion?

**Comment by pielco11** [comment]
@nonameable because a few weeks ago everything was fine and the code haven't changed since then, just my opinion

Feel free to help us to find out what's going on!

**Comment by Qiegao1994** [comment]
Thank you for your great code! I have the same question. I used twint -s  to get all the posts about the keywords. It looks nice at first. But it can only get the data around 20 days. I tried several times, and every time, I got the similar outcome-around 20 days contained 10000+posts. 
Then I tried --since 2019-12-26, but it still stop at 22/02/2020. (around 10000+posts)
And it showed the following information: 
**CRITICAL:root:twint.run:Twint:Feed:noDataExpecting value: line 1 column 1 (char 0)
CRITICAL:root:twint.run:Twint:Feed:Tweets_known_error:Expecting value: line 1 column 1 (char 0)
Expecting value: line 1 column 1 (char 0) [x] run.Feed**

Is that because of twitter restrictions? Is there any solution to solve this kind of issue?

Thank you for your great work again, looking forword to your reply!

**Comment by pielco11** [comment]
What happens if you select a strict window by time to time? Also 10k posts is a well result!

**Comment by Qiegao1994** [comment]
> What happens if you select a strict window by time to time? Also 10k posts is a good result!

Yes, 10k is really good. But what I really need is to get the posts since 26/12/2019. I think getting posts monthly is a good idea. I wanna know how to select the time? According to your 'CLI Basic Examples and Combos', I only see "year 2014" and "since", which cannot set a time period (from xxxx to xxxx). Is there any method to set the time period? Thank you for your prompt reply!

**Comment by pielco11** [comment]
`twint -s "keyword" --since "2019-01-01" --until "2019-02-01"`

**Comment by Qiegao1994** [comment]
> `twint -s "keyword" --since "2019-01-01" --until "2019-02-01"`

Thank you！I tried again and I found a new question. I used since "2020-01-20" --until "2020-02-20" (1 month), I got 10000+ posts and stopped. Then, I used "2020-02-19" --until "2020-02-20" (1 day), I got 10000+ posts and stopped too. So I think maybe there is a limitation from Twitter that it only provides 10000+ posts at most no matter the time period.

Anyway, thank you for your help!
